app/Http/Controllers/CustomerController.php

```diff
@@ -2,14 +2,109 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Favorite;
+use App\Models\Product;
+use App\Models\Purchase;
+use App\Models\User;
+use http\Exception;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
 
 class CustomerController extends Controller
 {
     //
+
+    private array $sort = ['sortByAsc', 'sortByDesc'];
     public function index()
     {
         //
     }
+
+    public function addFavoriteProduct($productId){
+        try {
+            $favorite = new Favorite();
+
+            $favorite->user_id = Auth::user()->id;
+            $favorite->product_id = $productId;
+
+            $favorite->save();
+
+            return redirect()->back();
+        }
+        catch (\Exception $e){
+            return redirect()->back();
+        }
```

Comment:
Waarom staat dit in try catch? is er niet iets af te vangen in een mooie foutmelding en zonder de try catch?